tests/test_preview.py

```diff
@@ -0,0 +1,706 @@
+    )
+    ensemble.entities = []
+    params = {"THERMO": [10, 20], "STEPS": [20, 30]}
+    ensemble = exp.create_ensemble(
+        "dir_test",
+        params=params,
+        run_settings=RunSettings("python", exe_args="sleep.py"),
+    )
+    gen_dir = fileutils.get_test_conf_path(osp.join("generator_files", "test_dir"))
+    symlink_dir = fileutils.get_test_conf_path(
+        osp.join("generator_files", "to_symlink_dir")
+    )
+    copy_dir = fileutils.get_test_conf_path(osp.join("generator_files", "to_copy_dir"))
+
+    ensemble.attach_generator_files()
+    ensemble.attach_generator_files(
+        to_configure=[gen_dir, copy_dir], to_copy=copy_dir, to_symlink=symlink_dir
+    )
+    preview_manifest = Manifest(ensemble)
+
+    # Call preview renderer for testing output
+    output = previewrenderer.render(exp, preview_manifest)
+
+    # Evaluate output
+    assert "Tagged Files for model configuration" in output
+    assert "Copy files" in output
+    assert "Symlink" in output
+    assert "Ensemble parameters" in output
+    assert "Model parameters" in output
+
+    assert "generator_files/test_dir" in output
+    assert "generator_files/to_copy_dir" in output
+    assert "generator_files/to_symlink_dir" in output
+    assert "/SmartSim" in output
+    for model in ensemble:
+        assert "generator_files/test_dir" in model.files.tagged[0]
+        for copy in model.files.copy:
+            assert "generator_files/to_copy_dir" in copy
+        for link in model.files.link:
+            assert "generator_files/to_symlink_dir" in link
+
+
+def test_preview_colocated_db_model_ensemble(fileutils, test_dir, wlmutils, mlutils):
+    """Test preview of DBModel on colocated ensembles, first adding the DBModel to the
+    ensemble, then colocating DB.
+    """
+
+    exp_name = "test-colocated-db-model-ensemble-reordered"
+    test_launcher = wlmutils.get_test_launcher()
+    test_interface = wlmutils.get_test_interface()
+    test_port = wlmutils.get_test_port()
+    test_device = mlutils.get_test_device()
+    test_num_gpus = 1
+
+    test_script = fileutils.get_test_conf_path("run_tf_dbmodel_smartredis.py")
+
+    exp = Experiment(exp_name, launcher=test_launcher, exp_path=test_dir)
+    colo_settings = exp.create_run_settings(exe=sys.executable, exe_args=test_script)
+    colo_settings.set_nodes(1)
+    colo_settings.set_tasks(1)
+
+    # Create the ensemble of two identical SmartSim Model
+    colo_ensemble = exp.create_ensemble(
+        "colocated_ens", run_settings=colo_settings, replicas=2
+    )
+
+    # Create colocated SmartSim Model
+    colo_model = exp.create_model("colocated_model", colo_settings)
+
+    # Create and save ML model to filesystem
+    content = "empty test"
+    path = pathlib.Path(test_dir) / "model1.pt"
+    with open(path, "w", encoding="utf-8") as prev_file:
+        prev_file.write(content)
+
+    model_path = test_dir + "/model1.pt"
```

Comment:
same path as `path` above? intentional? can we re-use it w/`str(path)` as arg to `.add_ml_model`?